src/components/Forms/BaseInput.jsx

```diff
@@ -9,7 +10,13 @@ const sizes = {
     large: "px-5 py-3.5 text-md leading-md", // 50px
 };
 
-export const BaseInput = ({ as: Tag, size = "medium", isError, className, isRequired, ...rest }) => {
+export const BaseInput = ({ as: Tag, size = "medium", isError, className, isRequired, value, ...rest }) => {
+    // added regexp to return only non-whitespace characters for strings
+    const stringValue = isString(value) && value.replace(/[^.\S]+/g, "");
```

Comment:
what will this return if there are white spaces in the text. for eg. "Sample text" what will this regex replace will return ?

Reply:
if the value is "Sample Text", then "SampleText" is returned.
But we only use this variable to test for an empty string and don't use it as an input value.